src/Sarif/Visitors/SarifVersionOneToCurrentVisitor.cs

```diff
@@ -206,24 +260,105 @@ public static LogicalLocation CreateLogicalLocation(LogicalLocationVersionOne v1
+        {
+            if (_currentRun.LogicalLocations == null)
             {
-                notifications = new List<Notification>();
+                _currentRun.LogicalLocations = new Dictionary<string, LogicalLocation>();
+            }
+
+            string fullyQualifiedName = logicalLocation.FullyQualifiedName;
+            string logicalLocationKey = logicalLocation.FullyQualifiedName;
+            int disambiguator = 0;
 
-                foreach (NotificationVersionOne notification in v1Notifications)
+            // We need to do the comparison here too
+            while (_currentRun.LogicalLocations.ContainsKey(logicalLocationKey))
+            {
+                LogicalLocation logLoc = _currentRun.LogicalLocations[logicalLocationKey].DeepClone();
+                logLoc.Name = GetLogicalLocationName(fullyQualifiedName);
+                logLoc.FullyQualifiedName = logicalLocation.FullyQualifiedName;
+
+                if (logicalLocation.ValueEquals(logLoc))
                 {
-                    notifications.Add(CreateNotification(notification));
+                    break;
                 }
+
+                logicalLocationKey = logicalLocation.FullyQualifiedName + "-" + disambiguator.ToString(CultureInfo.InvariantCulture);
+                disambiguator++;
+            }
+
+            if (!_currentRun.LogicalLocations.ContainsKey(logicalLocationKey))
+            {
+                _currentRun.LogicalLocations.Add(logicalLocationKey, logicalLocation);
+                FixUpLogicalLocation(logicalLocationKey);
+            }
+
+            return logicalLocationKey;
+        }
+
+        internal string GetLogicalLocationName(string fullyQualifiedLogicalName)
+        {
+            if (string.IsNullOrWhiteSpace(fullyQualifiedLogicalName))
+            {
+                throw new ArgumentNullException(nameof(fullyQualifiedLogicalName));
+            }
+
+            return fullyQualifiedLogicalName.Split(Utilities.FullyQualifiedNameDelimiters,
+                                                   StringSplitOptions.RemoveEmptyEntries).Last();
+        }
+
+        internal void FixUpAllLogicalLocations()
+        {
+            foreach (string key in _currentRun.LogicalLocations.Keys)
+            {
+                FixUpLogicalLocation(key);
+            }
+        }
+
+        internal void FixUpLogicalLocation(string key)
+        {
+            LogicalLocation logicalLocation = _currentRun.LogicalLocations[key];
+
+            if (logicalLocation.FullyQualifiedName == key)
+            {
+                logicalLocation.FullyQualifiedName = null;
+            }
+
+            if (logicalLocation.Name == key)
+            {
+                logicalLocation.Name = null;
+            }
+        }
+
+        internal Message CreateMessage(string text)
+        {
+            Message message = null;
+
+            if (!string.IsNullOrWhiteSpace(text))
```

Comment:
Tough call, but it somebody took the trouble to set `text` to `"   \t "`, I'd just go with it.

Reply:
Done